File: server/app/schemas/conversation.py
from datetime import datetime
from pydantic import BaseModel


class ConversationSummary(BaseModel):
    id: str
    title: str
    status: str
    priority: str
    assignee: str | None
    unreadCount: int
    lastMessagePreview: str
    lastMessageAt: datetime


class ConversationDetail(BaseModel):
    id: str
    title: str
    status: str
    priority: str
    assignee: str | None
    customerId: str
    unreadCount: int
    lastMessagePreview: str
    lastMessageAt: datetime
    createdAt: datetime
    updatedAt: datetime


class PaginatedConversations(BaseModel):
    items: list[ConversationSummary]
    page: int
    pageSize: int
    total: int


class StatusUpdateRequest(BaseModel):
    status: str


class PriorityUpdateRequest(BaseModel):
    priority: str

File: app/lib/features/inbox/models/conversation.dart
class ConversationSummary {
  ConversationSummary({
    required this.id,
    required this.title,
    required this.status,
    required this.priority,
    required this.assignee,
    required this.unreadCount,
    required this.lastMessagePreview,
    required this.lastMessageAt,
    bool? isTyping,
    this.presenceState,
  }) : isTyping = isTyping ?? false;

  final String id;
  final String title;
  final String status;
  final String priority;
  final String? assignee;
  final int unreadCount;
  final String lastMessagePreview;
  final DateTime lastMessageAt;

  final bool isTyping;
  final String? presenceState;

  factory ConversationSummary.fromJson(Map<String, dynamic> json) {
    return ConversationSummary(
      id: json['id'] as String,
      title: json['title'] as String,
      status: json['status'] as String,
      priority: json['priority'] as String,
      assignee: json['assignee'] as String?,
      unreadCount: json['unreadCount'] as int? ?? 0,
      lastMessagePreview: json['lastMessagePreview'] as String? ?? '',
      lastMessageAt: DateTime.parse(json['lastMessageAt'] as String),
      isTyping: false, // important
    );
  }

  ConversationSummary copyWith({
    String? title,
    String? status,
    String? priority,
    String? assignee,
    int? unreadCount,
    String? lastMessagePreview,
    DateTime? lastMessageAt,
    bool? isTyping,
    String? presenceState,
  }) {
    return ConversationSummary(
      id: id,
      title: title ?? this.title,
      status: status ?? this.status,
      priority: priority ?? this.priority,
      assignee: assignee ?? this.assignee,
      unreadCount: unreadCount ?? this.unreadCount,
      lastMessagePreview: lastMessagePreview ?? this.lastMessagePreview,
      lastMessageAt: lastMessageAt ?? this.lastMessageAt,
      isTyping: isTyping ?? this.isTyping,
      presenceState: presenceState ?? this.presenceState,
    );
  }
}
